homeassistant/components/sensor/tibber.py

```diff
@@ -38,16 +28,13 @@
 async def async_setup_platform(hass, config, async_add_entities,
                                discovery_info=None):
     """Set up the Tibber sensor."""
-    import tibber
-    tibber_connection = tibber.Tibber(config[CONF_ACCESS_TOKEN],
-                                      websession=async_get_clientsession(hass))
-
-    async def _close(*_):
-        await tibber_connection.rt_disconnect()
-    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, _close)
+    tibber_connection = hass.data.get(DOMAIN)
+    if tibber_connection is None:
```

Comment:
Make a guard clause above instead and check if discovery_info is None and return if so.